### DISCUSSION TITLE
ValueObservation not called when adding rows in App extension

### DISCUSSION THREAD

**Discussion by Jasperav** [groue/GRDB.swift#1264] [Q&A]
My app relies on the ValueObservations to display the newest data. It doesn't update when a different Database is making the commits (I am doing commits in the app extension). This is due to the fact there are multiple Databases and I understand a ValueObservation is attached to a Database, that's why it won't see the changes.

I am wondering if someone knows how other applications are dealing with this. Do they use ValueObservations at all? Do they refresh all observations each time (and if so, how?)?

The App extension I use is Push notifications. I want to update my database when the notification comes in.

**Comment by groue** [comment]
I hope users will share their technique.

But basically, you're correct: your app must restart the observation when it gets a message from the other process that tells "I did something to the database!"

There are starting points at https://github.com/groue/GRDB.swift/blob/master/Documentation/SharingADatabase.md#how-to-perform-cross-process-database-observation
